podium-gateway/src/main/java/nl/thehyve/podium/service/RequestService.java

```diff
@@ -219,7 +236,39 @@ private void init() {
         return getCountsForOrganisations(organisationUuids);
     }
 
+    /**
+     * Submit the request by uuid.
+     *
+     * @param user the current user, submitting the request
+     * @param uuid the uuid of the request
+     * @return the updated request
+     * @throws ActionNotAllowed if the request is not in status 'Revision'.
+     */
+    public RequestRepresentation submitRevision(AuthenticatedUser user, UUID uuid) throws ActionNotAllowed {
+        Request request = requestRepository.findOneByUuid(uuid);
 
+        log.debug("Access and status checks...");
+        AccessCheckHelper.checkRequester(user, request);
+        AccessCheckHelper.checkReviewStatus(request, RequestReviewStatus.Revision);
+
+        log.debug("Validating request data.");
+        RequestRepresentation requestData = requestMapper.requestToRequestDTO(request);
+        validateRequest(requestData);
```

Comment:
First you need to copy revisionDetail to requestDetail here, otherwise the contents of revisionDetail is not validated.